clients/go/datacatalog/client.go

```diff
@@ -0,0 +1,112 @@
+package datacatalog
+
+import (
+	"context"
+	"crypto/x509"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/flyteorg/flyteidl/clients/go/datacatalog/mocks"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/datacatalog"
+
+	"github.com/flyteorg/flytestdlib/logger"
+	grpcRetry "github.com/grpc-ecosystem/go-grpc-middleware/retry"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+)
+
+/*
+This package provides a basic wrapper for the data catalog client. There are two ways to
+create a data catalog client:
+1. Add a config section in the config file, like:
+```
+data-catalog:
+	endpoint: localhost:8089
+	inscure: true
+```
+Then in your application you can create a data catalog client like this:
+```
+client := datacatalog.InitializeDataCatalogClientFromConfig(ctx)
+client.GetArtifact(ctx, getArtifactRequest)
+```
+2. Call InitializeDataCatalogClient directly and fill the config in your application
+```
+cfg := datacatalog.Config{
+    Endpoint: "localhost:8089",
+    Insecure: false,
+}
+client := datacatalog.InitializeDataCatalogClient(ctx, cfg)
+client.GetArtifact(ctx, request)
+```
+*/
+
+var (
+	once                  = sync.Once{}
+	dataCatalogConnection *grpc.ClientConn
+)
+
+// Create a new Datacatalog client for task execution caching
+func NewDataCatalogClient(ctx context.Context, conn *grpc.ClientConn) (datacatalog.DataCatalogClient,
+	error) {
+
+	logger.Infof(ctx, "Initialized Admin client")
+	return datacatalog.NewDataCatalogClient(conn), nil
+}
+
+func NewDataCatalogConnection(ctx context.Context, cfg Config) (*grpc.ClientConn, error) {
```

Comment:
This code looks a whole lot like [this](https://github.com/flyteorg/flyteidl/blob/master/clients/go/admin/client.go#L188). Can we do some simple refactor to avoid building the connection in different ways?